mmrazor/structures/subnet/candidate.py

```diff
@@ -41,23 +49,37 @@ def __init__(self, initdata: Optional[Any] = None):
     @property
     def scores(self) -> List[float]:
         """The scores of candidates."""
-        return [item[1] for item in self.data]
+        return [
+            value.get('score', 0.) for item in self.data
+            for _, value in item.items()
+        ]
+
+    def resources(self, key_indicator: str = 'flops') -> List[float]:
+        """The resources of candidates."""
+        assert key_indicator in ['flops', 'params', 'latency']
+        return [
+            value.get(key_indicator, 0.) for item in self.data
+            for _, value in item.items()
+        ]
 
     @property
     def subnets(self) -> List[Dict]:
         """The subnets of candidates."""
-        return [item[0] for item in self.data]
+        return [eval(key) for item in self.data for key, _ in item.items()]
 
-    def _format(self, data: Any) -> _format_return:
-        """Transform [any, ...] to [tuple(any, float), ...] Transform any to
-        tuple(any, float)."""
+    def _format(self, data: Any):
```

Comment:
update type-hint for data with `Union[List[Dict], Dict].`

Reply:
> update type-hint for data with `Union[List[Dict], Dict].`

done